### PR TITLE
Issues with plotting functions

### PR DESCRIPTION
**Pull request by clementstross**
This pull request addresses
Issue #40 
Issue #38 
Issue #37 
Issue #36 

These were mostly identified on review by @CRSlade1, some were noticed by @clementstross himself. 

A large change was that the PrecisionRecall metrics now do NOT calculate the area under the curve. They have been aligned with sklearn.

The PDP and ALE plots now use a function for bucketing data.

### CODE REVIEW

File: R/null_gini.R [hunk 1]
```diff
@@ -39,6 +40,10 @@ null_gini <- function(actual, predicted, weight=NULL){
   # Combine actuals and predictions
   data <- data.frame(actual=actual, weight=weight, predicted=predicted)
 
+  if (na.rm==TRUE){
+    data <- data %>% filter(!is.na(actual) & !is.na(weight) & !is.na(predicted))
```
Comment (gloverd2) [inline R/null_gini.R:44]: dplyr::filter

File: R/plot_PDP.R [hunk 2]
```diff
@@ -148,78 +118,27 @@ plot_PDP <- function(data, model, explain_col, weight=NULL, n_bins=10, use_plotl
...
-
-
-    # Initialise plotting data
-    plot.data <- data.frame(start.x=rep(explain_bins_edges.x[1:n_bins.x], rep =n_bins.y), end.x=rep(explain_bins_edges.x[2:(n_bins.x+1)], rep =n_bins.y), center.x=rep(explain_bins_center.x, rep =n_bins.y),
-                            start.y=rep(explain_bins_edges.y[1:n_bins.y], each=n_bins.x), end.y=rep(explain_bins_edges.y[2:(n_bins.y+1)], each=n_bins.x), center.y=rep(explain_bins_center.y, each=n_bins.x)) %>%
-      dplyr::mutate(width.x=end.x-start.x,
-             width.y=end.y-start.y)
+    plot.data <- full_join(plot.data.x %>% select(lower, upper, center, width) %>% mutate(join=1),
+                           plot.data.y %>% select(lower, upper, center, width) %>%  mutate(join=1),
+                           by="join",
```
Comment (gloverd2) [inline R/plot_PDP.R:132]: dplyr::mutate

File: R/plot_PDP.R [hunk 3]
```diff
@@ -148,78 +118,27 @@ plot_PDP <- function(data, model, explain_col, weight=NULL, n_bins=10, use_plotl
...
-    }
-
-
-
-    # Initialise plotting data
-    plot.data <- data.frame(start.x=rep(explain_bins_edges.x[1:n_bins.x], rep =n_bins.y), end.x=rep(explain_bins_edges.x[2:(n_bins.x+1)], rep =n_bins.y), center.x=rep(explain_bins_center.x, rep =n_bins.y),
-                            start.y=rep(explain_bins_edges.y[1:n_bins.y], each=n_bins.x), end.y=rep(explain_bins_edges.y[2:(n_bins.y+1)], each=n_bins.x), center.y=rep(explain_bins_center.y, each=n_bins.x)) %>%
-      dplyr::mutate(width.x=end.x-start.x,
-             width.y=end.y-start.y)
+    plot.data <- full_join(plot.data.x %>% select(lower, upper, center, width) %>% mutate(join=1),
```
Comment (gloverd2) [inline R/plot_PDP.R:130]: dplyr::full_join & dplyr::mutate

File: R/metric_ROC.R [hunk 4]
```diff
@@ -19,7 +19,7 @@
 #'
 #' metric_ROC(actual=data$target, predicted=data$x1)
 #' metric_ROC(actual=data$target, predicted=data[c("x1","x2")])
-metric_ROC <- function(actual, predicted, weight=NULL){
+metric_ROC <- function(actual, predicted, weight=NULL, na.rm=FALSE){
```
Comment (gloverd2) [inline R/metric_ROC.R:22]: na.rm is not documented

File: R/null_gini.R [hunk 5]
```diff
@@ -22,7 +23,7 @@
 #'
 #' null_gini(actual, predicted, weight)
 #'
-null_gini <- function(actual, predicted, weight=NULL){
+null_gini <- function(actual, predicted, weight=NULL, na.rm=FALSE){
```
Comment (gloverd2) [inline R/null_gini.R:26]: na.rm is not documented

File: R/null_ROC.R [hunk 6]
```diff
@@ -13,6 +13,7 @@
 #' Threshold - value of the threshold
 #' TP_Rate - True positive rate
 #' FP_Rate - False positive rate
+#' @export
```
Comment (gloverd2) [inline R/null_ROC.R:16]: na.rm is not documented

**Review by gloverd2 — CHANGES_REQUESTED** [review]
1. are there tests for plotting numerical buckets?  

2. Optionally, can we think of a name for this function that agrees with the naming conventions of other functions i.e. prep_ ? 

3. I can see some changes have been made to remove certain packagename::'s from functions such as ggplot2::aes. I understand why it's been done, however it is causing errors if we test the package using RStudio. I do realise it doesnt stop us compiling the package however I feel as though using the test package function of RStudio benefits us in a lot of ways. (It picks up on undocumented params)

**Comment by clementstross** [comment]
> 1. are there tests for plotting numerical buckets?
> 2. Optionally, can we think of a name for this function that agrees with the naming conventions of other functions i.e. prep_ ?
> 3. I can see some changes have been made to remove certain packagename::'s from functions such as ggplot2::aes. I understand why it's been done, however it is causing errors if we test the package using RStudio. I do realise it doesnt stop us compiling the package however I feel as though using the test package function of RStudio benefits us in a lot of ways. (It picks up on undocumented params)

Lets talk about names for this function in the next meeting. We might also want to make this function internal so not export it.

**Review by gloverd2 — APPROVED** [review]
I believe my latest changes were just to do with package namespaces. Otherwise I think these functions will be popular as we found in the example codebase doc they are pretty cool.

**Pull request merged**